renderdoc/os/posix/android/android_stringio.cpp

```diff
@@ -63,15 +64,33 @@ namespace FileIO
 {
 const char *GetTempRootPath()
 {
-  return "/sdcard";
+  static string ret;
+  GetExecutableFilename(ret);
+
+  // This folder is writable even if the APK does not have manifest write permissions.
+  ret = "/data/data/" + ret + "/files";
+  return ret.c_str();
 }
 
+// For RenderDocCmd.apk, this returns "org.renderdoc.renderdoccmd"
+// For other APKs, we use it to get the writable temp directory.
 void GetExecutableFilename(string &selfName)
 {
-  char path[512] = {0};
-  readlink("/proc/self/exe", path, 511);
+  char buf[4096];
+  snprintf(buf, sizeof(buf), "/proc/%u/cmdline", getpid());
```

Comment:
Command line apps behave just like they do on Linux, but they are the exception (mainly for developers). App APKs always contain shared libraries which dont get an argv, and I think if you got it by other means it would just say "/system/bin/app_process64", which is what the /proc/self/exe Linux code returned.